indigo_api/models/works.py

```diff
@@ -595,6 +617,11 @@ def expressions(self):
     def can_delete(self):
         return not self.expressions().exists()
 
+    def save(self):
```

Comment:
Are you trying to prevent this object from being created in certain circumstances?

Doing it like this is dangerous because `save()` will simply not happen in some cases, and silently at that. It breaks the contract of `Model.save()` and will have unforeseen impacts on other code.

If we need to prevent the creation, it should be part of a validation step wherever the creation happens (eg. in a form or view). It's too late to be doing it here because we can't stop it cleanly.

Reply:
Cool, I've canned this for now, it's not necessary. We probably don't want consolidations and amendments on the same date, but that's actually a separate issue which we can deal with if and when it actually arises.